src/gs-overview-page.c

```diff
@@ -486,8 +486,20 @@ get_categories_cb (GObject *source_object,
 	gs_shell_side_filter_clear_categories (priv->shell);
 
 	for (guint i = 0; i < list->len; i++) {
+		gboolean usb_category_inserted = FALSE;
 		cat = GS_CATEGORY (g_ptr_array_index (list, i));
-		if (gs_category_get_size (cat) == 0)
+
+		if (g_strcmp0 (gs_category_get_id (cat), "usb") == 0) {
+			g_autoptr(GList) copy_dests = gs_plugin_loader_dup_copy_dests (plugin_loader);
+
+			if (copy_dests != NULL && g_list_length (copy_dests) > 0)
```

Comment:
Do we need to use the `g_list_length` here to test for whether the list has an element? i.e. can the list have no elements but not be `NULL`? From that function's doc:
`To check whether the list is non-empty, it is faster to check list against NULL.`

Reply:
You're right. It's easy to forget that `GList*` is a unique snowflake where the default state of the object is not actually being an object :smile:

Fixed locally.